src/shared/rpc/IPCSocketClient.cc

```diff
@@ -117,23 +117,36 @@ IPCSocketClient ::send(std::string_view data)
 }
 
 IPCSocketClient::ReadStatus
-IPCSocketClient::read_all(std::string &content)
+IPCSocketClient::read_all(std::string &content, std::chrono::milliseconds timeout_ms, int attempts)
 {
   if (this->is_closed()) {
     // we had a failure.
     return {};
```

Comment:
I know it's not part of the changes, but this will return `NO_ERROR`? But the comment indicates a failure.  Could `UNKNOWN` be a better return?